vortex-array/src/stats/statsset.rs

```diff
@@ -188,12 +188,16 @@ impl IntoIterator for StatsSet {
     }
 }
 
+impl FromIterator<(Stat, Scalar)> for StatsSet {
```

Comment:
We should still go through the set function to avoid creating incorrect stats vecs